src/cppcache/src/ThinClientRegion.hpp

```diff
@@ -416,14 +416,21 @@ class ChunkedFunctionExecutionResponse : public TcrChunkedResult {
 
  public:
   inline ChunkedFunctionExecutionResponse(
-      TcrMessage& msg, bool getResult, ResultCollectorPtr rc,
-      ACE_Recursive_Thread_Mutex* resultCollectorLock = NULL)
+      TcrMessage& msg, bool getResult, ResultCollectorPtr rc)
       : TcrChunkedResult(),
         m_msg(msg),
         m_getResult(getResult),
-        m_rc(rc),
-        m_resultCollectorLock(resultCollectorLock) {}
-
+        m_rc(rc) {}
+
+    inline ChunkedFunctionExecutionResponse(
```

Comment:
There are two constructors now because the previous solitary constructor had the last parameter defaulted to `NULL`. Now that it's a smart pointer, I split the constructor into two pieces, one with the lock and one without.